### PR TITLE
 [Doc] delete .md in navigation page to avoid 404 in official web

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docs/introduction/StarRocks_intro.md [hunk 1]
```diff
@@ -10,20 +10,20 @@ Join our [Slack channel](https://join.slack.com/t/starrocks/shared_invite/zt-z5z
 <NavBoxPart title="About StarRocks">
 <NavBoxPartItem>
 
-- [Introduction](../introduction/what_is_starrocks.md)
-- [Architecture](../introduction/Architecture.md)
-- [Shared-data StarRocks cluster](../deployment/deploy_shared_data.md)
-- [Features](../introduction/Features.md)
+- [Introduction](../introduction/what_is_starrocks)
+- [Architecture](../introduction/Architecture)
+- [Shared-data StarRocks cluster](../deployment/deploy_shared_data)
```
Comment (evelynzhaojie) [inline docs/introduction/StarRocks_intro.md:15]: i think we can remove “StarRocks”

File: docs/introduction/StarRocks_intro.md [hunk 2]
```diff
@@ -62,41 +62,41 @@ Join our [Slack channel](https://join.slack.com/t/starrocks/shared_invite/zt-z5z
...
 <NavBoxPartItem title="Query acceleration">
 
-- [Cost-based optimizer](../using_starrocks/Cost_based_optimizer.md)
-- [Synchronous materialized view](../using_starrocks/Materialized_view-single_table.md)
-- [Asynchronous materialized views](../using_starrocks/Materialized_view.md)
-- [Colocate Join](../using_starrocks/Colocate_join.md)
-- [Query cache](../using_starrocks/query_cache.md)
+- [Cost-based optimizer](../using_starrocks/Cost_based_optimizer)
+- [Synchronous materialized view](../using_starrocks/Materialized_view-single_table)
+- [Asynchronous materialized views](../using_starrocks/Materialized_view)
```
Comment (evelynzhaojie) [inline docs/introduction/StarRocks_intro.md:67]: the singular form "view" is used for sync mv, why the plural form "views" is used for async mv

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by evelynzhaojie — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.1

**Pull request merged**